app/javascript/dashboard/api/ApiClient.js

```diff
@@ -3,9 +3,25 @@
 const API_VERSION = `/api/v1`;
 
 class ApiClient {
-  constructor(url) {
+  constructor(modelName, options = {}) {
     this.apiVersion = API_VERSION;
-    this.url = `${this.apiVersion}/${url}`;
+    this.options = options;
+    this.modelName = modelName;
```

Comment:
Does `resource` sound like a better name for `modelName`, which is bit backend specific ?